### PR TITLE
Store location of last opened file. Closes #38

### LINKED ISSUE #38: File opener doesn't remember last location
Hi.

When opening a file, I navigate to the folder my file is located. The file is then played in gnome-mpv, great!

I have another file in the exact same folder, so I press Ctrl-O to open it: I have to navigate AGAIN to the same folder.

I think it would be great if it remembered my last location so I could just pick my new file in that same directory.

Thanks !

### PR DESCRIPTION
**Pull request by gnarula**

### CODE REVIEW

**Comment by gnome-mpv** [comment]
Thank you, but I think this should be made optional since it is overriding default gtkfilechooser behaviour.

File: src/actionctl.c [hunk 1]
```diff
@@ -78,6 +92,11 @@ static void open_handler(	GSimpleAction *action,
 
 	file_chooser = GTK_FILE_CHOOSER(open_dialog);
 
+	if(filepath)
```
Comment (TingPing) [inline src/actionctl.c]: This should check for an empty string not NULL.

File: src/actionctl.c [hunk 2]
```diff
@@ -62,9 +64,21 @@ static void open_handler(	GSimpleAction *action,
...
 	GtkWidget *open_dialog;
+	GSettings *config;
+	GSettingsBackend *config_backend;
+
+	config_backend = g_keyfile_settings_backend_new
+				(	config_file,
+					CONFIG_ROOT_PATH,
+					CONFIG_ROOT_GROUP );
+
+	config = g_settings_new_with_backend(CONFIG_WIN_STATE, config_backend);
```
Comment (TingPing) [inline src/actionctl.c]: This should be unref'd

**Comment by TingPing** [comment]
Sort-of unrelated to this PR really but @gnome-mpv why do you force the keyfile settings backend on everyone...?, You should use the default backend which can be overridden by platform or user.

File: src/actionctl.c [hunk 3]
```diff
@@ -78,6 +92,11 @@ static void open_handler(	GSimpleAction *action,
 
 	file_chooser = GTK_FILE_CHOOSER(open_dialog);
 
+	if(filepath)
+	{
+		gtk_file_chooser_set_current_folder(file_chooser, dirname(filepath));
```
Comment (TingPing) [inline src/actionctl.c]: `g_path_get_dirname()` to be more portable.

**Comment by gnome-mpv** [comment]
@TingPing: I didn't know gsettings backend could be overriden by the user, and I thought having a plain text config file would be convenient. Is it possible to override the backend on a per-application basis?

**Comment by TingPing** [comment]
@gnome-mpv Every GIO extension point can be overridden by an env var. In this case `GSETTINGS_BACKEND=key-file` should work. Obviously it isn't really designed to be set per-application, but there is no reason for it to be either.

**Comment by gnarula** [comment]
@gnome-mpv In that case, should I add a boolean preference to override the default behaviour?
@TingPing thanks for the suggestions! Will correct them and update the PR.

**Comment by gnome-mpv** [comment]
>In that case, should I add a boolean preference to override the default behaviour?

Yes.

**Comment by gnome-mpv** [comment]
@TingPing: Okay. I'll switch to using the default backend.

**Comment by gnarula** [comment]
@gnome-mpv Made it optional. Please have a look now.

**Comment by gnarula** [comment]
Updated to store directory path instead of storing the filepath. Used g_clear_object() to unref config objects. Sorry if this is taking a while... I'm quite new to GTK

**Comment by gnome-mpv** [comment]
Merged as 8595154.

**Pull request closed without merging**